lib/exandra.ex

```diff
@@ -327,6 +327,39 @@ defmodule Exandra do
 
     {keyspace, conn}
   end
+
+  @doc """
+  Streams the results of a simple query or a prepared query to the given callback.
+  The argument will be a `Xandra.PageStream` that you can iterate on to process
+  each `Xandra.Page and accumulate any results you need.
+
+  See `Xandra.prepare!/4` and `Xandra.stream_pages!/4` for more information including
+  supported options.
+
+  ## Examples
+
+      Exandra.stream!(
+        {"SELECT * FROM USERS WHERE can_contact = ?", [true]},
+        MyRepo,
+        fn page ->
+          Enum.each(stream, fn page ->
+            Enum.each(page, &send_email(&1.email, "Hello!")
+          end)
+        end
+      )
+  """
+  @spec stream!({String.t(), list(term())}, atom(), function(), Keyword.t()) :: term()
+  def stream!({sql, values}, repo, callback, opts \\ []) do
+    repo.checkout(fn conn ->
+      conn
+      |> @xandra_mod.stream_pages!(
+        @xandra_mod.prepare!(conn, sql, opts),
+        values,
+        opts
+      )
+      |> callback.()
+    end)
```

Comment:
I’m thinking: if we exposed an (internal) function to check out the **cluster** from the repo (instead of a connection), we could do `Xandra.Cluster.stream_pages!/4`, which picks the right node anyways, and would remove the need for a callback. Thoughts?

Reply:
I like the idea of using cluster instead so it won't have to keep a single connection for the duration of the function. I'll work on this.